### DISCUSSION TITLE
Energy (not power) reporting

### DISCUSSION THREAD

**Discussion by thd6n75** [PeteRager/lennoxs30#264] [Ideas]
It would be nice if this plugin produced energy (as in, "power over time", as in kWh) sensors in addition to the existing power (unit of measurement == watts, kilowatts) sensors.

I enabled the diagnostic sensors, and see that I have an entity named "System_inverter_energy / sensor.system_inverter_energy" on my heat pump device.  However, that entity reports an instantaneous **power** measurement in watts - it is not an **energy** (kWh) sensor, despite the name.  It can not be used by Home Assistant's built-in Energy dashboard-  see https://www.home-assistant.io/docs/energy/individual-devices/#devices-with-power-w-sensors .  It requires additional effort to create a Riemann sum sensor ( https://www.home-assistant.io/integrations/integration/#energy ) in order to get a real energy sensor.  It would be nice if the integration did this for us.

**Comment by PeteRager** [comment]
Yes that would be nice.  Lennox does not provide that information.  In general the purpose of the integration is to make the data from Lennox available to be further shaped by using HA.  So most of it is pass through.  That said, the one piece of data that is calculated is the inverted power - the data Lennox provides is the inverter current and the inverter voltage - so in this case it was simple to multiply these together and get the watts.  

The docs currently describe what to do, if those are incorrect then we should update.

https://github.com/PeteRager/lennoxs30#inverter-power

There is a lot of code to do the Riemann sum correctly and handle all the edge cases - sensor unavailable, restore, etc.

https://github.com/home-assistant/core/blob/dev/homeassistant/components/integration/sensor.py

**Comment by thd6n75** [comment]
Shoot, I was looking in the diagnostics docs.  Sorry I missed it.
